job-dsl-plugin/build.gradle

```diff
@@ -48,12 +52,12 @@ jenkinsPlugin {
 
 dependencies {
     compile project(':job-dsl-core')
-    optionalJenkinsPlugins 'org.jenkins-ci.plugins:cloudbees-folder:4.2@jar'
+    optionalJenkinsPlugins 'org.jenkins-ci.plugins:cloudbees-folder:4.4@jar'
     optionalJenkinsPlugins 'org.jenkins-ci.plugins:credentials:1.9.4@jar'
     optionalJenkinsPlugins 'org.jenkins-ci.plugins:vsphere-cloud:1.1.11@jar'
     optionalJenkinsPlugins 'org.jenkins-ci.plugins:config-file-provider:2.7@jar'
     jenkinsTest 'org.jenkins-ci.plugins:ant:1.2@jar' // see JENKINS-17129
-    jenkinsTest 'org.jenkins-ci.main:maven-plugin:1.509.3@jar'
+    jenkinsTest "org.jenkins-ci.main:maven-plugin:${jenkinsVersion}@jar"
```

Comment:
After 1.534 the maven-plugin got it's own release cycle, so there is no version 1.554.1. See http://repo.jenkins-ci.org/releases/org/jenkins-ci/main/maven-plugin/. Not sure why that didn't cause any problems?!?

I extracted the core update to #345. I will fix this pull request when merging.